src/documents/models.py

```diff
@@ -267,17 +277,90 @@ def __str__(self):
         return str(created)
 
     @property
-    def source_path(self):
+    def source_filename(self):
+        if self.filename is None:
+            self.filename = self.generate_source_filename()
+
+        return self.filename
+
+    @staticmethod
+    def many_to_dictionary(field):
+        # Converts ManyToManyField to dictionary by assuming, that field
+        # entries contain an _ or - which will be used as a delimeter
```

Comment:
Just a minor nit: it is "delimiter", not "delimeter". Feel free to correct it if you want to, but it's no big deal. :slightly_smiling_face:

Reply:
Yep, thanks for the feedback - fixed.